### PR TITLE
Updated Noir-CLI to support Nargo >= v0.31.0

### PR DESCRIPTION
**Pull request by DanieleDiBenedetto**

### CODE REVIEW

File: src/bin/noir_cli/cli.rs [hunk 1]
```diff
@@ -57,17 +69,37 @@ pub enum Commands {
         #[arg(long)]
         output_pubs: Option<PathBuf>,
     },
+    /// Parse proof as generated by bb tool from Noir toolchain and output files for proof and
+    /// public inputs suitable to be submitted to zkVerify. Both in hex and binary format.
+    /// To be used for Noir versions >= 0.31.0.
+    ProofDatav2 {
```
Comment (DanieleDiBenedetto) [inline src/bin/noir_cli/cli.rs:75]: If you can find a better name I would be grateful :D

**Review by lgiussan — APPROVED** [review]
LGTM!

**Review by rushby — APPROVED** [review]
LGTM

**Review by la10736 — APPROVED** [review]
LGTM

**Pull request merged**